packages/radix-ui-themes/src/components/radio.tsx

```diff
@@ -0,0 +1,48 @@
+'use client';
+
+import * as React from 'react';
+import classNames from 'classnames';
+import { composeEventHandlers } from '@radix-ui/primitive';
+import { composeRefs } from '@radix-ui/react-compose-refs';
+import { radioPropDefs } from './radio.props.js';
+import { extractProps } from '../helpers/index.js';
+import { marginPropDefs } from '../props/index.js';
+
+import type { ComponentPropsWithoutColor, NotInputRadioAttributes } from '../helpers/index.js';
+import type { MarginProps, GetPropDefTypes } from '../props/index.js';
+
+type RadioElement = React.ElementRef<'input'>;
+type RadioOwnProps = GetPropDefTypes<typeof radioPropDefs> & {
+  value: string;
+  onValueChange?: (value: string) => void;
```

Comment:
`onValueChange` for a single radio doesn't really make sense, it's only really makes sense from a radio group pov.

It's more to do with its `checked` state, so should be rather `defaultChecked`/`checked`/`onCheckedChange` (or `onCheck` given that it can only really be checked and never unchecked).

Reply:
Let's chat about this, that's how I initially thought about it too until I tried to implement that

Reply:
Decided that it makes sense with how the native implementation reports values instead of the checked state